client.go

```
@@ -259,13 +259,14 @@ func sendRequestStream[T streamable](client *Client, req *http.Request, retryOpt
 		}
 
 		// handle status codes
-		failures = append(failures, fmt.Sprintf("#%d/%d error response received: %v", i+1, options.Retries+1, client.handleErrorResp(resp)))
+		errResp := client.handleErrorResp(resp)
+		failures = append(failures, fmt.Sprintf("#%d/%d error response received: %v", i+1, options.Retries+1, errResp))
 
 		// exit on non-retriable status codes
 		if !options.canRetry(resp.StatusCode) {
 			failures = append(failures, fmt.Sprintf("exiting due to non-retriable error in try #%d/%d: %d %s", i+1, options.Retries+1, resp.StatusCode, resp.Status))
 			slog.Error("sendRequestStream failed due to non-retriable statuscode", "code", resp.StatusCode, "status", resp.Status, "tries", i+1, "maxTries", options.Retries+1, "failures", strings.Join(failures, "; "))
-			return nil, fmt.Errorf("request failed on non-retriable status-code: %d %s", resp.StatusCode, resp.Status)
+			return nil, fmt.Errorf("request failed on non-retriable status-code: %s", resp.StatusCode, errResp.Error())
```

I feel like we should just return the error. `request failed on non-retriable status-code:` seems like internal details that shouldn't get exposed to the user.

---

Not sure.. this is like the lowest level we get to. Maybe rather have some error unwrapping logic some layers up?

---

yeah, not sure. It also has other stuff in front of the call so it can't return the exact error until we fix it all the way up.